src/Docker.DotNet/Endpoints/SwarmOperations.cs

```diff
@@ -130,6 +129,15 @@ async Task<IEnumerable<SwarmService>> ISwarmOperations.ListServicesAsync(Cancell
             return this._client.JsonSerializer.DeserializeObject<SwarmService[]>(response.Body);
         }
 
+        async Task<IEnumerable<SwarmService>> ISwarmOperations.ListServicesAsync(ServicesListParameters parameters, CancellationToken cancellationToken)
+        {
+            IQueryString queryParameters = new QueryString<ServicesListParameters>(parameters);

```

Comment:
Can you validate this against null and skip the query parameters if it is. Then update the original `ListServicesAsync` to just call this so we don't have two impls.

Reply:
Agreed.